runtime/src/snapshot_utils.rs

```diff
@@ -1246,32 +1246,12 @@ pub fn add_bank_snapshot(
 ) -> Result<BankSnapshotInfo> {
     let mut add_snapshot_time = Measure::start("add-snapshot-ms");
     let slot = bank.slot();
-    // bank_snapshots_dir/slot
     let bank_snapshot_dir = get_bank_snapshot_dir(&bank_snapshots_dir, slot);
-    if bank_snapshot_dir.is_dir() {
-        // There is a time window from when a snapshot directory is created to when its content
-        // is fully filled to become a full state good to construct a bank from.  At the init time,
-        // the system may not be booted from the latest snapshot directory, but an older and complete
-        // directory.  Then, when adding new snapshots, the newer incomplete snapshot directory could
-        // be found.  If so, it should be removed.
-        purge_bank_snapshot(&bank_snapshot_dir)?;
-    } else {
-        // Even the snapshot directory is not found, still ensure the account snapshot directory
-        // is also clean.  hardlink failure will happen if an old file exists.
-        let account_paths = &bank.accounts().accounts_db.paths;
-        let slot_str = slot.to_string();
-        for account_path in account_paths {
-            let account_snapshot_path = account_path
-                .parent()
-                .ok_or(SnapshotError::InvalidAccountPath(account_path.clone()))?
-                .join("snapshot")
-                .join(&slot_str);
-            if account_snapshot_path.is_dir() {
-                // remove the account snapshot directory
-                move_and_async_delete_path(&account_snapshot_path);
-            }
-        }
-    }
+    assert!(
```

Comment:
This assertion makes sense to me in the context of a running validator.

In terms of ledger-tool, I think it is valid with the current way we do things; but thinking there's a potential edge-case in the future. As we discussed on another PR (#31921), the eventual goal is that ledger-tool will unarchive into the same dirs as we normally use, and then do rebuilding from snapshot the same way as fast-boot.

It's possible for ledger-tool to create a snapshot at the same slot it loaded from, in which case the snapshot is not incomplete nor old, and thus wouldn't be purged at startup.

wdyt?